client/upe-onboarding-wizard/upe-preview-methods-selector/add-payment-methods-task.js

```diff
@@ -1,33 +1,139 @@
 import { __ } from '@wordpress/i18n';
-import React, { useCallback, useContext, useState } from 'react';
+import React, {
+	useCallback,
+	useContext,
+	useState,
+	useEffect,
+	useMemo,
+} from 'react';
 import { Button, Card, CardBody, ExternalLink } from '@wordpress/components';
 import interpolateComponents from 'interpolate-components';
 import WizardTaskContext from '../wizard/task/context';
 import CollapsibleBody from '../wizard/collapsible-body';
 import WizardTaskItem from '../wizard/task-item';
-import { useGetAvailablePaymentMethodIds } from '../../data';
+import {
+	useGetAvailablePaymentMethodIds,
+	useEnabledPaymentMethodIds,
+	useSettings,
+} from '../../data';
 import PaymentMethodCheckboxes from '../../components/payment-methods-checkboxes';
 import PaymentMethodCheckbox from '../../components/payment-methods-checkboxes/payment-method-checkbox';
+import LoadableSettingsSection from 'wcstripe/settings/loadable-settings-section';
 
-const AddPaymentMethodsTask = () => {
-	const availablePaymentMethods = useGetAvailablePaymentMethodIds();
+const usePaymentMethodsCheckboxState = () => {
+	const [ initialEnabledPaymentMethodIds ] = useEnabledPaymentMethodIds();
+	const [ paymentMethodsState, setPaymentMethodsState ] = useState( {} );
+
+	useEffect( () => {
+		setPaymentMethodsState(
+			initialEnabledPaymentMethodIds.reduce(
+				( map, paymentMethod ) => ( {
+					...map,
+					[ paymentMethod ]: true,
+				} ),
+				{}
+			)
+		);
+	}, [ initialEnabledPaymentMethodIds, setPaymentMethodsState ] );
+
+	const handleChange = useCallback(
+		( paymentMethodName, enabled ) => {
+			setPaymentMethodsState( ( oldValues ) => ( {
+				...oldValues,
+				[ paymentMethodName ]: enabled,
+			} ) );
+		},
+		[ setPaymentMethodsState ]
+	);
+
+	return [ paymentMethodsState, handleChange ];
+};
+
+const ContinueButton = ( { paymentMethodsState } ) => {
 	const { setCompleted } = useContext( WizardTaskContext );
-	const [ paymentMethodsState, setPaymentMethodsState ] = useState( {
-		card: true,
-	} );
+	const [
+		initialEnabledPaymentMethodIds,
+		updateEnabledPaymentMethodIds,
+	] = useEnabledPaymentMethodIds();
+
+	const { saveSettings, isSaving } = useSettings();
+
+	const checkedPaymentMethods = useMemo(
+		() =>
+			Object.entries( paymentMethodsState )
+				.map( ( [ method, enabled ] ) => enabled && method )
+				.filter( Boolean ),
+		[ paymentMethodsState ]
+	);
+
+	const unCheckedPaymentMethods = Object.entries( paymentMethodsState )
+		.map( ( [ method, enabled ] ) => ! enabled && method )
+		.filter( Boolean );
```

Comment:
Quick Question: why `useMemo` is not used for `unCheckedPaymentMethods` similar as `checkedPaymentMethods`?

Reply:
Yeah, I guess it makes sense to use it in both :D Thanks! Adjusted in ba66bfc1